### ISSUE TITLE
Clock::SetMode has been set multi-place

### ISSUE THREAD

**Issue by xmyqsh** [ApolloAuto/apollo#9158]
The `Clock::SetMode` has been set in localization([rtk](https://github.com/ApolloAuto/apollo/blob/master/modules/localization/rtk/rtk_localization_component.cc#L29), [msf](https://github.com/ApolloAuto/apollo/blob/master/modules/localization/msf/msf_localization_component.cc#L33), [ndt](https://github.com/ApolloAuto/apollo/blob/master/modules/localization/ndt/ndt_localization_component.cc#L34)),  and [planning](https://github.com/ApolloAuto/apollo/blob/master/modules/planning/planning_component.cc#L52) 

When the `FLAGS_use_sim_time` is enabled, there will be one module either localization or planning running with the wrong Clock.

In other words, the `FLAGS_use_sim_time` mode does not support the whole stack simulation currently.
